File: app/src/main/java/com/mastip/personaltodolist/screen/activity/base/MyBaseActivity.java
package com.mastip.personaltodolist.screen.activity.base;

import android.os.Bundle;
import android.os.PersistableBundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.widget.Toast;

import com.mastip.personaltodolist.utilities.Constant;
import com.mastip.personaltodolist.utilities.CustomSharedPreferences;

import butterknife.ButterKnife;

/**
 * Created by HateLogcatError on 8/28/2017.
 */

public abstract class MyBaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getContentViewId());
        ButterKnife.bind(this);
        new CustomSharedPreferences().setSharedPreferences(getSharedPreferences(Constant.TAG_SHARED_PREFERENCE, MODE_PRIVATE));
        initiateItem();
    }

    public void showToast(String message) {
        Toast.makeText(this, "" + message, Toast.LENGTH_SHORT).show();
    }

    public abstract int getContentViewId();
    public abstract void initiateItem();
}

File: app/src/main/java/com/mastip/personaltodolist/screen/activity/base/MyBaseActivity.kt
package com.mastip.personaltodolist.screen.activity.base

import android.content.Context
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.widget.Toast
import butterknife.ButterKnife
import com.mastip.personaltodolist.utilities.Constant
import com.mastip.personaltodolist.utilities.CustomSharedPreferences

abstract class MyBaseActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(contentViewId)
        ButterKnife.bind(this)
        CustomSharedPreferences.sharedPreferences = getSharedPreferences(Constant.TAG_SHARED_PREFERENCE, Context.MODE_PRIVATE)
        initiateItem()
    }

    fun showToast(message: String) {
        Toast.makeText(this, "" + message, Toast.LENGTH_SHORT).show()
    }

    abstract val contentViewId: Int
    abstract fun initiateItem()
}